tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/RecordTypes.fs

```diff
@@ -116,7 +116,7 @@ module RecordTypes =
             (Warning 464, Line 15, Col 22, Line 15, Col 28, "This code is less generic than indicated by its annotations. A unit-of-measure specified using '_' has been determined to be '1', i.e. dimensionless. Consider making the code generic, or removing the use of '_'.")
             (Warning 464, Line 15, Col 35, Line 15, Col 42, "This code is less generic than indicated by its annotations. A unit-of-measure specified using '_' has been determined to be '1', i.e. dimensionless. Consider making the code generic, or removing the use of '_'.")
             (Error 5, Line 17, Col 1, Line 17, Col 5, "This field is not mutable")
-            (Error 1, Line 17, Col 16, Line 17, Col 22, "The type 'decimal<Kg>' does not match the type 'float<Kg>'")
+            (Error 1, Line 17, Col 16, Line 17, Col 22, "The type 'float<Kg>' does not match the type 'decimal<Kg>'")
```

Comment:
Let's simplify this case a bit to

<img width="2079" height="416" alt="Image" src="https://github.com/user-attachments/assets/b3138260-6a7a-4d86-95d0-18710c2f07f3" />

The situation is similar here, the expected type is explicitly specified and is `decimal<Kg>`, but the error message threats it as the actual type.

Also note that the error range is on the second argument, not on the operator.  This is due to the fact that in the error 
`ErrorFromAddingTypeEquation(...ConstraintSolverTypesNotInEqualityRelation(...m1), m2)`the wrong m is taken. But I would fix it in a separate PR.